### PR TITLE
Rate limit bigquery dataset loading

### PR DESCRIPTION
**Pull request by paco-valdez**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

TBD

**Description of Changes Made (if issue reference is not provided)**

When connecting to a bigquery db if a dataset contains a lot of tables, Cube will hit the rate limit for concurrent requests.

### CODE REVIEW

**Comment by pacofvf** [comment]
I set the rate limit to 10, although I'm not sure what is the ideal limit.

**Comment by jediwarpraptor** [comment]
> @jediwarpraptor this is the change I had in mind, it looks like I need to add type hints. (I'm new to TS). The other proposed solution from Pavel was to add an env variable to only read from that dataset.

@pacofvf - I absolutely think this would fix the current issue I'm seeing in my environment.  Because I use BQ datasets liberally as the primary means of organization in the warehouse to separate/segregate based on both information domain (subscriptions, retail, wholesale, etc.) as well as ELT transformation stage (ingestion/source, staging, final, etc.), for my use-case (and others like mine), issuing hundreds of API calls through one big `.map()` immediately exceeds Google's rate limits.

re: the tweaking of that parameter, my gut tells me that `10` is a reasonable default. I don't know how easy it is within your codebase to abstract that to a config value which might be more performant in other conditions.  

My gut tells me that since this kind of metadata discovery will not need to be run often (at least during regular usage); combined with the fact that anyone who has fewer BigQuery datasets are unlikely to notice/perceive any performance slowdown related to this change.

tl;dr; Looks good to me! 
👍

**Comment by jediwarpraptor** [comment]
Also apologies for not responding sooner, I put this project on hold until the beginning of the year and am only just coming back to it now.

**Comment by paco-valdez** [comment]
We added a map-reduce logic on this logic, so it should be fixed in latest release.
